### ISSUE TITLE
2.Calculul metricilor: precision, recall, accuracy, PR-AUC. Ștefania(Gr.B)

### ISSUE THREAD

**Issue by licastefania200001-alt** [georgerusu017/churn_rate_predicition#50]

**Comment by licastefania200001-alt** [comment]
import pandas as pd
from sklearn.metrics import accuracy_score, precision_score, recall_score, average_precision_score

df = pd.read_csv("tranzactii.csv", sep="\t")

df = df.dropna(subset=['Class'])

df['Class'] = df['Class'].astype(int)

y_true = df['Class']

y_score = df[['f1','f2','f3','f4','f5']].sum(axis=1)

threshold = 0.5
y_pred = (y_score >= threshold).astype(int)

print("Accuracy :", accuracy_score(y_true, y_pred))
print("Precision:", precision_score(y_true, y_pred, zero_division=0))
print("Recall:", recall_score(y_true, y_pred, zero_division=0))
print("PR-AUC:", average_precision_score(y_true, y_score))

**Comment by licastefania200001-alt** [comment]
import pandas as pd
from sklearn.metrics import accuracy_score, precision_score, recall_score, average_precision_score

df = pd.read_csv("tranzactii.csv", sep="\t")

df = df.dropna(subset=['Class'])

df['Class'] = df['Class'].astype(int)

y_true = df['Class']

y_score = df[['f1','f2','f3','f4','f5']].sum(axis=1)

threshold = 0.5
y_pred = (y_score >= threshold).astype(int)

print("Accuracy :", accuracy_score(y_true, y_pred))
print("Precision:", precision_score(y_true, y_pred, zero_division=0))
print("Recall:", recall_score(y_true, y_pred, zero_division=0))
print("PR-AUC:", average_precision_score(y_true, y_score))
